bridgesample/src/androidTest/java/com/livefront/bridgesample/scenario/activity/LargeDataBackstackActivityTest.kt

```diff
@@ -0,0 +1,95 @@
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.Espresso.pressBack
+import androidx.test.espresso.action.ViewActions.*
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.ext.junit.rules.ActivityScenarioRule
+import androidx.test.platform.app.InstrumentationRegistry
+import com.livefront.bridgesample.R
+import com.livefront.bridgesample.scenario.util.getCurrentActivity
+import org.junit.Assert.assertEquals
+import org.junit.Rule
+
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class LargeDataBackstackActivityTest {
+
+    @get:Rule
+    var activityScenarioRule = ActivityScenarioRule<LargeDataActivity>(
+            LargeDataActivity.getNavigationIntent(
+                    InstrumentationRegistry.getInstrumentation().targetContext,
+                    true
+            )
+    )
+
+    @Test
+    fun generateDataAndNavigateForward() {
+        for (i in 1..5) {
+            onView(withId(R.id.generateDataButton)).perform(click())
+            onView(withId(R.id.navigateButton)).perform(click())
+        }
+
+        // Wait 1 second to allow a crash to occur
+        Thread.sleep(1000)
+    }
+
+    @Test
+    fun generateDataAndNavigateForwardAndBackward() {
+        onView(withId(R.id.generateDataButton)).perform(click())
+        var originalBitmap: Bitmap?
+        (getCurrentActivity() as LargeDataActivity).apply {
+            originalBitmap = this.savedBitmap
+        }
+        onView(withId(R.id.navigateButton)).perform(click())
+
+        for (i in 1 until 5) {
+            onView(withId(R.id.generateDataButton)).perform(click())
+            onView(withId(R.id.navigateButton)).perform(click())
+        }
+
+        for (i in 1..5) {
+            pressBack()
+        }
+
+        (getCurrentActivity() as LargeDataActivity).apply {
+            assertEquals(originalBitmap, this.savedBitmap)
+        }
+
+        // Wait 1 second to allow a crash to occur
+        Thread.sleep(1000)
+    }
+
+    @Test
+    fun generateDataAndNavigateForwardRotateBackward() {
+        onView(withId(R.id.generateDataButton)).perform(click())
+        var originalBitmap: Bitmap? = null
+        (getCurrentActivity() as LargeDataActivity).apply {
+            originalBitmap = this.savedBitmap
+        }
+        onView(withId(R.id.navigateButton)).perform(click())
+
+        for (i in 1 until 5) {
+            onView(withId(R.id.generateDataButton)).perform(click())
+            onView(withId(R.id.navigateButton)).perform(click())
+        }
+
+        UiDevice.getInstance(InstrumentationRegistry.getInstrumentation()).setOrientationLeft()
+
+        for (i in 1..5) {
+            pressBack()
+        }
+
+        (getCurrentActivity() as LargeDataActivity).apply {
+            assertEquals(originalBitmap, this.savedBitmap)
+        }
+
+        // Wait 1 second to allow a crash to occur
+        Thread.sleep(1000)
+    }
+}
```

Comment:
Let's make sure to add newlines to all these files.

Reply:
Done.